### PR TITLE
Ratings page: anime-level list + client-side statistics (v0.14.12)

### PR DESCRIPTION
**Pull request by Priapos1004**
## Summary

Builds the real `/ratings` page (was a navbar placeholder): a tabbed route (**Ratings** + **Statistics**, `?tab=` like `/admin`) driven **entirely client-side from one extended `GET /ratings/scores` fetch** shared by both tabs — no per-user stats endpoint, no scipy. The Ratings tab is an anime-level list (score-band cover grid + sortable table) with genre/age/season filters; the Statistics tab is summary, score histogram, You-vs-MAL alignment, genre/studio breakdown, attribute correlations, and activity charts. Removes the dead `/statistics` navbar placeholder.

## Changes

- **Backend (minimal):** extend the `RatingScoreItem` projection (cover, `mal_score`, `scored_by`, `episodes`, `total_watch_time`, season, `created_at`, `relation_type`) — all on already-eager-loaded `media → anime` rows, so **zero added query cost**; the DAO query is unchanged.
- **Frontend:** new `lib/components/ratings/` (tab nav, list tab → band grid / table / filter bar, stats tab → 5 chart sections), pure stats math in `lib/utils/ratingStats.ts` (closed-form weighted least-squares, Spearman ρ, eta), `ratingsFilter` store, `chartTheme.ts` (shared `chartTooltipStyle`), and a shared `SegmentedControl`.
- **Charts:** every chart re-mounts on section entry so its entrance animation replays — backed by a generic `EChart` fix (hold the first `setOption` until the first measured `ResizeObserver` callback).
- **Post-review hardening** (last two commits): fixed a null-MAL sort bug, HTML-escaped catalog strings in 4 ECharts tooltips (incl. a pre-existing `RatingsOverviewTimeline` sink), epoch-based + unparseable-date-safe activity time math, non-negative clamps, `SEASON_ORDER` consolidation, dead-code removal.

## Quality Checks (ISO 25010)

- [x] Linting passed — `ruff check .` clean
- [x] Tests passed — backend `pytest` 696 passed; frontend `vitest` 225 passed (+ new regression tests); `svelte-check` 0 errors / 0 warnings
- [x] **Functional Suitability** — null-MAL/Δ sort bug found in review and fixed (+ regression tests); stats math (WLS/Spearman/eta/histogram) verified total and edge-case-safe (empty / single / all-equal)
- [x] **Performance Efficiency** — backend zero-added-query-cost confirmed; client stats are O(n)/O(n log n) inside `$derived`; date sorts decorate-once; ECharts instances dispose and only the active section mounts
- [x] **Reliability** — fetch failure / 401-403 / empty-state handled; unparseable `created_at` dropped; `log10` / watch-time sums clamped ≥ 0
- [x] **Security** — `/ratings/scores` is user-scoped (no cross-user leak), no new unauthed endpoint, no secrets; catalog strings in tooltip HTML now `escapeHtml`'d (the one injection sink), incl. the pre-existing timeline
- [x] **Maintainability** — dead `getUserRole` removed; `SEASON_ORDER` consolidated to one source; no orphaned utils; new code unit-tested
- [x] **Flexibility** — charts use centralized theme tokens / palettes; titles resolve via the name-language helper; list scores honor the `rating_step` setting
- [x] UI verified — built iteratively in-browser with the maintainer, section by section

## Compound Doc

See: `compound-docs/2026-06-23-v0.14.12-ratings-page.md`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by Priapos1004 — COMMENTED** [review]
basst

File: phsar/frontend/src/lib/utils/ratingStats.ts [hunk 1]
```diff
@@ -0,0 +1,654 @@
...
+	if (!values.length) return [];
+	const w = Math.max(1, Math.min(Math.round(window), values.length));
+	const out: number[] = [];
+	for (let i = 0; i < values.length; i++) {
+		const start = Math.max(0, i - w + 1);
+		const slice = values.slice(start, i + 1);
+		out.push(slice.reduce((a, b) => a + b, 0) / slice.length);
+	}
+	return out;
+}
```
[the remark this answers is not in the document]
Comment (Priapos1004) [reply phsar/frontend/src/lib/utils/ratingStats.ts:614]: Fixed in fb3ea30: rewrote `movingAverage` as a rolling sum (add the entering value, drop the one leaving the trailing window) — O(n) with no per-point slice/reduce allocation. Output is unchanged and guarded by the existing `movingAverage([2,4,6,8],2) → [2,3,5,7]` test.

File: phsar/frontend/src/lib/utils/navigation.ts [hunk 2]
```diff
@@ -33,6 +33,13 @@ export function buildDetailHref(
     return `/${type}?${params.toString()}`;
 }
 
+/** Jump to an anime-view search filtered to a single studio ("other anime from this
+ * studio"). Shared by StudioLinks and the ratings genre/studio chart so the filter
+ * shape can't drift between them. */
+export function searchByStudio(studio: string): void {
+	void navigateToSearch({ query: '', search_type: 'title', view_type: 'anime', studio_name: [studio] });
+}
```
[the remark this answers is not in the document]
Comment (Priapos1004) [reply phsar/frontend/src/lib/utils/navigation.ts:41]: Fixed at the source in fb3ea30: `navigateToSearch` now logs non-401 failures (`console.error`) instead of re-throwing, so it never rejects. That covers `searchByStudio` plus every other fire-and-forget caller (the home + search pages) in one place — no caller awaits it or relies on the throw.
